Sources/StreamChatUI/Utils/ImageLoading/NukeImageLoader.swift

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import Nuke
+import UIKit
+
+extension ImageTask: Cancellable {}
+
+/// The class which is resposible for loading images from URLs.
+/// Internally uses `Nuke`'s shared object of `ImagePipeline` to load the image.
+open class NukeImageLoader: ImageLoading {
+    @discardableResult
+    open func loadImage(
+        using urlRequest: URLRequest,
+        cachingKey: String,
```

Comment:
```suggestion
        cachingKey: String?,
```
cachingKey should be optional, but even better would be to have an `ImageLoadingOptions` just like `ImageRequestOptions`. Otherwise, whenever we add more options, the arguments of the func will increase a lot

Reply:
Good point. I have made the `cachingKey` optional for now. As it is a single option currently, not introducing `ImageLoadingOptions`. We can introduce this later if needed

Reply:
Introducing it later will cause a breaking change. And by introducing it now, we will avoid breaking the API in the future, since the `ImageLoadingOptions` should be a struct with all of the props with default arguments. Currently `cachingKey` is optional, but doesn't have a default value. So ideally we want `struct ImageLoadingOptions` and with all props with default arguments, in this case: `ImageLoadingOptions(cachingKey: String? = nil)`